web/src/beta/features/WorkspaceSetting/innerPages/Workspaces/index.tsx

```diff
@@ -21,17 +21,17 @@ type Props = {
 const WorkspaceSetting: FC<Props> = ({ workspace }) => {
   const t = useT();
 
-  const [localWorkspaceName, setLocalWorkspaceName] = useState<string>(
-    workspace?.name ?? ""
-  );
-
   const { useUpdateWorkspace: updateWorkspace } = useWorkspaceFetcher();
-  const handleSubmitUpdateWorkspaceName = useCallback(async () => {
-    await updateWorkspace({
-      teamId: workspace?.id ?? "",
-      name: localWorkspaceName
-    });
-  }, [workspace, localWorkspaceName, updateWorkspace]);
+
+  const handleSubmitUpdateWorkspaceName = useCallback(
+    async (name: string) => {
+      await updateWorkspace({
+        teamId: workspace?.id ?? "",
+        name
+      });
+    },
+    [workspace, updateWorkspace]
+  );
```

Comment:
- There's no follow up actions, so maybe we don't need async await in this case.
- Better return directly if no workspace?.id instead of calling api with empty workspace id.